packages/twenty-server/src/engine/twenty-orm/pg-shared-pool/__tests__/pg-shared-pool.service.spec.ts

```diff
@@ -0,0 +1,351 @@
+  | 'LOG_LEVELS';
+
+// Define return types for config values to avoid 'unknown' error
+type ConfigValue = boolean | number | LogLevel[] | string;
+
+// Define pool interface matching our implementation
+interface PoolWithEndTracker extends Pool {
+  __hasEnded?: boolean;
+}
+
+// Mock the pg module
+jest.mock('pg', () => {
+  const mockPool = jest.fn().mockImplementation(() => ({
+    on: jest.fn(),
+    end: jest.fn().mockImplementation((callback) => {
+      if (callback) callback();
+
+      return Promise.resolve();
+    }),
+    _clients: [],
+    _idle: [],
+    _pendingQueue: { length: 0 },
+  }));
+
+  mockPool.prototype = {
+    on: jest.fn(),
+    end: jest.fn(),
+  };
+
+  return {
+    Pool: mockPool,
+  };
+});
+
+describe('PgPoolSharedService', () => {
+  let service: PgPoolSharedService;
+  let configService: TwentyConfigService;
+  let mockLogger: Partial<Logger>;
+
+  const configValues: Record<ConfigKey, ConfigValue> = {
+    PG_ENABLE_POOL_SHARING: true,
+    PG_POOL_MAX_CONNECTIONS: 10,
+    LOG_LEVELS: ['error', 'warn'],
+  };
+
+  beforeEach(async () => {
+    mockLogger = {
+      log: jest.fn(),
+      debug: jest.fn(),
+      warn: jest.fn(),
+      error: jest.fn(),
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        PgPoolSharedService,
+        {
+          provide: TwentyConfigService,
+          useValue: {
+            // Type cast for the mock function to match TwentyConfigService
+            get: jest
+              .fn()
+              .mockImplementation(
+                (key: string) => configValues[key as ConfigKey],
+              ),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<PgPoolSharedService>(PgPoolSharedService);
+    configService = module.get<TwentyConfigService>(TwentyConfigService);
+
+    // Replace the logger with our mock
+    (service as any).logger = mockLogger;
+  });
+
+  afterEach(() => {
+    // Unpatch the service and clean up its state AFTER each test
+    if (service && typeof (service as any).unpatchForTesting === 'function') {
+      (service as any).unpatchForTesting();
+    }
+    jest.clearAllMocks();
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('initialize', () => {
+    it('should initialize and patch pg Pool when enabled', () => {
+      service.initialize();
+
+      expect(mockLogger.log).toHaveBeenCalledWith(
+        expect.stringContaining(
+          'Pool sharing will use max 10 connections per pool',
+        ),
+      );
+      expect(mockLogger.log).toHaveBeenCalledWith(
+        expect.stringContaining('Pg pool sharing initialized'),
+      );
+    });
+
+    it('should not initialize when pool sharing is disabled', () => {
+      jest
+        .spyOn(configService, 'get')
+        .mockImplementation((key: string): ConfigValue => {
+          if (key === 'PG_ENABLE_POOL_SHARING') return false;
```

Comment:
i think we can improve typing here, looks like TwentyConfig service is properly typed